test/test_indexing.py

```diff
@@ -458,6 +459,457 @@ def test_reduction_tensor_descriptor_indexing_reduction_loop(self):
         torch.testing.assert_close(result, expected)
         self.assertExpectedJournal(code)
 
+    def test_2d_slice_index(self):
+        """Test both setter from scalar and getter for [:,i]"""
+
+        @helion.kernel
+        def kernel(
+            src: torch.Tensor, dst: torch.Tensor
+        ) -> tuple[torch.Tensor, torch.Tensor]:
+            N = src.shape[1]
+            for i in hl.grid(N):
+                dst[:, i] = 1.0  # Test setter with scalar
+                src[:, i] = dst[:, i]  # Test getter from dst and setter to src
+            return src, dst
+
+        N = 128
+        src = torch.zeros([1, N], device=DEVICE)
+        dst = torch.zeros([1, N], device=DEVICE)
+
+        src_result, dst_result = kernel(src, dst)
+
+        # Both should be ones after the kernel
+        expected_src = torch.ones([1, N], device=DEVICE)
+        expected_dst = torch.ones([1, N], device=DEVICE)
+        torch.testing.assert_close(src_result, expected_src)
+        torch.testing.assert_close(dst_result, expected_dst)
+
+    @skipIfNormalMode(
+        "AssertionError in roll_reduction.py:104 - stored_node is not a torch.fx.Node"
+    )
```

Comment:
I have an upcoming PR stack that will fix all `skipIfNormalMode`-skipped tests in this PR.